File: weatherAPI/weather.py
import requests

from settings import BASE_URL, API_KEY


def getData(city: str, units: str = 'metric') -> dict:
    url = f'{BASE_URL}q={city}&appid={API_KEY}&units={units}'
    response = requests.get(url)
    if response.status_code == 200:
        return response.json()
    else:
        raise Exception(response.status_code)
    
def getWeather(city: str, units: str = 'metric') -> dict:
    data = getData(city=city, units=units)
    return data['main']

File: backend/weather.ts
const BASE_URL = 'https://api.openweathermap.org/data/2.5/weather';
const API_KEY = '141467b3afe32bf42206016b8de79bda';
const UNITS = 'metric';

export default async function getWeather(lat: Number, lon: Number): Promise<Object> {
    const res = await fetch(`${BASE_URL}?lat=${lat}&lon=${lon}&appid=${API_KEY}&units=${UNITS}`);
    return await res.json()['main'];
}